engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/types/EspTypeUtils.scala

```diff
@@ -135,14 +133,14 @@ object EspTypeUtils {
 
   private def extractGenericReturnType(typ: Type): Option[TypingResult] = {
     typ match {
-      case t: ParameterizedTypeImpl => extractGenericMonadReturnType(t)
+      case t: ParameterizedType => extractGenericMonadReturnType(t)
       case t => None
     }
   }
 
   // This method should be used only for method's and field's return type - for method's parameters such unwrapping has no sense
-  private def extractGenericMonadReturnType(genericReturnType: ParameterizedTypeImpl): Option[TypingResult] = {
-    val rawType = genericReturnType.getRawType
+  private def extractGenericMonadReturnType(genericReturnType: ParameterizedType): Option[TypingResult] = {
+    val rawType = genericReturnType.getRawType.asInstanceOf[Class[_]]
```

Comment:
I wonder - are we really sure it'll always be the case? Or maybe it'd be safer to write:
` case t: ParametrizedType if t.getRawType.isInstanceOf[Class[_]]` or sth like that?

Reply:
changed